File: data/data_transpiled/at_coder/abc121/D/4887155.java
public static String solve ( String string ) {
  int a = Integer . parseInt ( string ) ;
  int b = Integer . parseInt ( string ) ;
  if ( b == 0 ) {
    return "0" ;
  }
  String binA = a + "" ;
  String binB = b + "" ;
  int [ ] aa = new int [ 40 ] ;
  for ( int i = 0 ;
  i < aa . length ;
  i ++ ) {
    String _a = binA . substring ( i , i + 1 ) ;
    String _b = binB . substring ( i , i + 1 ) ;
    aa [ i ] = _a . equals ( "0" ) ? - 1 : Integer . parseInt ( binA . substring ( i + 1 , i + 1 ) , 2 ) ;
  }
  int [ ] bb = new int [ 40 ] ;
  for ( int i = 0 ;
  i < bb . length ;
  i ++ ) {
    String _b = binB . substring ( i , i + 1 ) ;
    bb [ i ] = _b . equals ( "0" ) ? - 1 : Integer . parseInt ( binB . substring ( i , i + 1 ) , 2 ) ;
  }
  int index = 0 ;
  do {
    if ( bb [ index ] != - 1 ) {
      break ;
    }
    index ++ ;
  }
  StringBuilder ans = new StringBuilder ( ) ;
  while ( index < 40 ) {
    if ( aa [ index ] == bb [ index ] == - 1 ) {
      ans . append ( "0" ) ;
    }
    else if ( aa [ index ] == - 1 ) {
      ans . append ( String . valueOf ( ( bb [ index ] + 1 ) % 2 ) ) ;
    }
    else if ( bb [ index ] == - 1 ) {
      ans . append ( String . valueOf ( aa [ index ] % 2 ) ) ;
    }
    else {
      ans . append ( String . valueOf ( ( bb [ index ] + 1 - aa [ index ] ) % 2 ) ) ;
    }
    index ++ ;
  }
  ans . setLength ( ans . length ( ) - 1 ) ;
  ans . append ( a % 2 == b % 2 == 0 ? String . valueOf ( ( b - a ) / 2 ) : String . valueOf ( ( ( b - a ) / 2 + 1 ) % 2 ) ) ;
  return String . valueOf ( Long . parseLong ( ans . toString ( ) , 2 ) ) ;
}

File: data/data_original/at_coder/abc121/D/4887155.py
def solve(string):
    a, b = map(int, string.split())
    if b == 0:
        return "0"
    bin_a = "{:040b}".format(a)
    bin_b = "{:040b}".format(b)
    aa = [-1 if _a == "0" else int(bin_a[i + 1:] or "0", 2) for i, _a in enumerate(bin_a)]
    bb = [-1 if _b == "0" else int(bin_b[i + 1:] or "0", 2) for i, _b in enumerate(bin_b)]
    index = 0
    while True:
        if bb[index] != -1:
            break
        index += 1
    ans = []
    while index < 40:
        if aa[index] == bb[index] == -1:
            ans.append("0")
        elif aa[index] == -1:
            ans.append(str((bb[index] + 1) % 2))
        elif bb[index] == -1:
            ans.append(str(aa[index] % 2))
        else:
            ans.append(str((bb[index] + 1 - aa[index]) % 2))
        index += 1
    ans[-1] = str((b - a) // 2 % 2) if a % 2 == b % 2 == 0 else str(((b - a) // 2 + 1) % 2)
    return str(int("".join(ans), 2))


if __name__ == '__main__':
    print(solve(input()))